### PR TITLE
First shot at networking player movement

### PR DESCRIPTION
**Pull request by spruceb**

### CODE REVIEW

File: Assets/Scripts/Managers/NetworkPlayerMappingManager.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    public override void OnJoinedRoom()
+    {
+        Utility.Print("Joined room", LogLevel.Error);
+        this.FrameDelayCall(() => OwnNextPlayer(PhotonNetwork.LocalPlayer.ActorNumber));
```
Comment (dkonik) [inline Assets/Scripts/Managers/NetworkPlayerMappingManager.cs]: why frame delay call? I feel like frame delay calls will cause issues and we should start phasing it out, is there a particular reason you used it here?

Comment (spruceb) [reply Assets/Scripts/Managers/NetworkPlayerMappingManager.cs]: I agree on phasing it out. I'm just not sure how to handle without it here. It seems like when I don't have it, the buffered RPC sometimes isn't sent before this is called so the state is all fucked.

File: Assets/Scripts/Managers/NetworkManager.cs [hunk 2]
```diff
@@ -75,4 +75,9 @@ public override void OnPlayerEnteredRoom(Photon.Realtime.Player newPlayer)
         base.OnPlayerEnteredRoom(newPlayer);
         Utility.Print("Another peep joined the room yo!", LogLevel.Error);
     }
+
+    public override void OnPlayerLeftRoom(Photon.Realtime.Player other)
+    {
+        Utility.Print("Another peep joined the room yo!", LogLevel.Error);
```
Comment (kjvic) [inline Assets/Scripts/Managers/NetworkManager.cs]: Can we add info to this? e.g. see https://doc-api.photonengine.com/en/pun/v2/class_photon_1_1_realtime_1_1_player-members.html

Comment (kjvic) [reply Assets/Scripts/Managers/NetworkManager.cs]: Also, this should say "Another peep *left* the room yo!"

File: Assets/Scripts/Managers/NetworkPlayerMappingManager.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    public bool LocalOwnsPlayer(int playerNumber)
+    {
+        if (playerOwners.ContainsKey(playerNumber))
+        {
+            return playerOwners[playerNumber] == PhotonNetwork.LocalPlayer.ActorNumber;
+        }
+        return false;
+    }
+
+    void OwnNextPlayer(int actorId)
```
Comment (kjvic) [inline Assets/Scripts/Managers/NetworkPlayerMappingManager.cs]: [nit] could you put `OwnNextPlayer_Internal` next to `OwnNextPlayer`?

File: Assets/Scripts/Managers/NetworkPlayerMappingManager.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        OwnNextPlayer_Internal(actorId);
+        photonView.RPC("SetPlayerState", RpcTarget.OthersBuffered, freePlayers.ToArray(), playerOwners);
+        foreach (var player in GameManager.instance.players) {
+            player.HandleOwnership();
+        }
+    }
+
+    void ReleaseAllPlayers(int actorId)
+    {
+        ReleaseAllPlayers_Internal(actorId);
```
Comment (kjvic) [inline Assets/Scripts/Managers/NetworkPlayerMappingManager.cs]: [nit] put `ReleaseAllPlayers_Internal` under `ReleaseAllPlayers`

File: Assets/Scripts/Managers/NetworkManager.cs (hunk 2 above)
Comment (spruceb) [reply Assets/Scripts/Managers/NetworkManager.cs]: gone

**Comment by dkonik** [comment]
I feel like `NetworkPlayerMappingManager` will end up having to manage more than just mappings. Maybe we should just make that be a `NetworkedPlayerManager`, have every player register itself with `NetworkedPlayerManager` on start. Then, when a room is joined, it can figure out which player will be the local player and call into that player which will then do the `MakeSelfLocal` (or whatever you called it)

**Comment by spruceb** [comment]
> I feel like `NetworkPlayerMappingManager` will end up having to manage more than just mappings. Maybe we should just make that be a `NetworkedPlayerManager`, have every player register itself with `NetworkedPlayerManager` on start. Then, when a room is joined, it can figure out which player will be the local player and call into that player which will then do the `MakeSelfLocal` (or whatever you called it)

Changed the name, but it already uses the GameModel's list of players. Don't really see a reason to keep a duplicate list?

**Pull request merged**